ResizeObserver shim type issues

**Bug**

When using `dojo/framework/shim/ResizeObserver` typescript is not able to find the correct typings

```
node_modules/@dojo/framework/shim/ResizeObserver.d.ts(2,20): error TS2304: Cannot find name 'ResizeObserverCallback'.
node_modules/@dojo/framework/shim/ResizeObserver.d.ts(2,45): error TS2304: Cannot find name 'ResizeObserver'.
node_modules/@dojo/framework/shim/ResizeObserver.d.ts(3,16): error TS2304: Cannot find name 'ResizeObserver'.
```

Possibly include and export the typings in the shim module.

--- a/src/shim/ResizeObserver.ts
+++ b/src/shim/ResizeObserver.ts
@@ -1,12 +1,40 @@
 import global from './global';
 import has from '../has/has';
 `!has('build-elide')`;
-import ResizeObserver from 'resize-observer-polyfill';
+import Resize from 'resize-observer-polyfill';
+
+export interface DOMRectReadOnly {
+	readonly x: number;
+	readonly y: number;
+	readonly width: number;
+	readonly height: number;
+	readonly top: number;
+	readonly right: number;
+	readonly bottom: number;
+	readonly left: number;
+}
+
+export interface ResizeObserverCallback {
+	(entries: ResizeObserverEntry[], observer: ResizeObserver): void;
+}
+
+export interface ResizeObserverEntry {
+	readonly target: Element;
+	readonly contentRect: DOMRectReadOnly;
+}
+
+export interface ResizeObserver {
+	prototype: ResizeObserver;
+	new (callback: ResizeObserverCallback): ResizeObserver;
+	observe(target: Element): void;
+	unobserve(target: Element): void;
+	disconnect(): void;
+}
 
 if (!has('build-elide')) {
 	if (!global.ResizeObserver) {
-		global.ResizeObserver = ResizeObserver;
+		global.ResizeObserver = Resize;
 	}
 }
 
-export default global.ResizeObserver as typeof ResizeObserver;
+export default global.ResizeObserver as ResizeObserver;

